source/ada/lsp-ada_handlers.adb

```diff
@@ -1808,8 +1817,131 @@ package body LSP.Ada_Handlers is
          end loop;
          Self.Server.On_Show_Message (Errors);
       end if;
+
+      --  Index the files immediately after loading a project
+      Self.Index_Files;
    end Load_Project;
 
+   -------------------------------
+   -- Get_Unique_Progress_Token --
+   -------------------------------
+
+   function Get_Unique_Progress_Token
+     (Self      : access Message_Handler;
+      Operation : String := "") return LSP_Number_Or_String
+   is
+
+      Pid : constant String :=
+        GNATCOLL.Utils.Image (Pid_To_Integer (Current_Process_Id), 1);
+   begin
+      Self.Token_Id := Self.Token_Id + 1;
+      --  Generate an identifier that has little risk of collision with
+      --  other language servers, or other occurrences of this server.
+      --  (There is still a very small risk of collision with PID recyclings,
+      --  but the consequences are acceptable.)
+      return
+        (Is_Number => False,
+         String    => To_LSP_String
+           ("ada_ls-"
+            & Pid & "-" & Operation & "-"
+            & GNATCOLL.Utils.Image (Self.Token_Id, 1)));
+   end Get_Unique_Progress_Token;
+
+   -----------------
+   -- Index_Files --
+   -----------------
+
+   procedure Index_Files (Self : access Message_Handler) is
+      type Context_And_Files is record
+         Context : Context_Access;
+         Sources : File_Array_Access;
+      end record;
+      type Context_And_Files_Array is array
+        (Natural range <>) of Context_And_Files;
+
+      Arr   : Context_And_Files_Array (1 .. Natural (Self.Contexts.Length));
+
+      token : constant LSP.Types.LSP_Number_Or_String
```

Comment:
'Token' instead of 'token'?

Reply:
For this I'm deliberately using the casing provided by the protocol, since this is a protocol object. But it's not at all a strong opinion :-)

Reply:
Ah I see, fine by me then, keep the protocol casing here